swift/ql/lib/codeql/swift/elements/decl/DeclWithMembers.qll

```diff
@@ -0,0 +1,26 @@
+private import codeql.swift.elements.decl.IterableDeclContext
+private import codeql.swift.elements.decl.VarDecl
+private import codeql.swift.elements.decl.SubscriptDecl
+private import codeql.swift.elements.decl.Decl
+
+class DeclWithMembers extends Decl, IterableDeclContext {
```

Comment:
I'm still of the opinion that there's not really a need to have a class like this as part of the public AST. Rather, I'd much rather see the importing of the `IterableDeclContext` class disappear from `swift.qll`, and its use purely as an implementation detail in order to get `getImmediateMember`/`getMember` predicates on the [classes](https://github.com/github/codeql/blob/a1edd655420b86a776c7a8c3659fbacd194ca664/swift/ql/lib/codeql/swift/generated/decl/NominalTypeDecl.qll#L9) [that](https://github.com/github/codeql/blob/a1edd655420b86a776c7a8c3659fbacd194ca664/swift/ql/lib/codeql/swift/generated/decl/ExtensionDecl.qll#L10) currently extend `IterableDeclContext`.

I think we should delay this change until we've settled on a decision here.

Reply:
And I'm not very happy with the name `DeclWithMembers`.  I think we're talking about a declaration of a kind that _can_ have members, not one that necessarily does.  Consider:
```
struct MyStruct {}
```